{
  try {
    callAsync(new Callable<Void>(){
      @Override public Void call() throws Exception {
        String subject=messages.getMessage("passwordWasResetMailMessageSubject",new Object[]{appName},Locale.ENGLISH);
        String text=messages.getMessage("passwordWasResetMailMessageBody",new Object[]{username,appBaseUrl,appName,password},Locale.ENGLISH);
        MimeMessage mimeMsg=kaaMessagingMailSender.createMimeMessage();
        MimeMessageHelper helper=new MimeMessageHelper(mimeMsg,"UTF-8");
        try {
          helper.setFrom(mailFrom);
          helper.setTo(email);
          helper.setSubject(subject);
          helper.setText(text,true);
          kaaMessagingMailSender.send(helper.getMimeMessage());
        }
 catch (        MessagingException e) {
          LOG.error("Unexpected error while sendPasswordAfterResetMail",e);
        }
        return null;
      }
    }
);
  }
 catch (  Exception e) {
    LOG.error("Unexpected error while sendPasswordAfterResetMail",e);
  }
}
